message/consumer.go

```diff
@@ -6,92 +6,94 @@ import (
 	"github.com/ONSdigital/go-ns/kafka"
 	"github.com/ONSdigital/go-ns/log"
+	"context"
 )
 
-//go:generate moq -out ./message_test/consumer_generated_mocks.go -pkg message_test . KafkaMessageConsumer KafkaMessage KafkaMessageProducer InsertedProducer
+//go:generate moq -out ./message_test/consumer_generated_mocks.go -pkg message_test . KafkaMessage KafkaConsumer CompletedProducer ErrorEventHandler
 
 const (
-	eventRecieved       = "Recieved DimensionsExtractedEvent"
+	eventRecieved       = "Recieved NewInstanceEvent"
 	eventKey            = "event"
-	eventHandlerErr     = "Unexpected error encountered while handling DimensionsExtractedEvent"
+	eventHandlerErr     = "Unexpected error encountered while handling NewInstanceEvent"
 	eventHandlerSuccess = "Instance has been successfully imported"
 	errorRecieved       = "Consumer exit channel recieved error. Exiting dimensionExtractedConsumer"
 	conumserErrMsg      = "Kafka Consumer Error recieved"
-	producerErrMsg      = "InsertedProducer Error recieved"
+	producerErrMsg      = "Completed Error recieved"
+	consumerStoppedMsg  = "Exiting Consumer loop"
+	unmarshallErrMsg    = "Unexpected error when unmarshalling avro message to newInstanceEvent"
 )
 
-// KafkaMessageConsumer type for consuming kafka messages.
-type KafkaMessageConsumer kafka.MessageConsumer
+var closer chan bool = make(chan bool)
 
-// KafkaMessageProducer type for producing kafka messages.
-type KafkaMessageProducer kafka.MessageProducer
-
-// kafkaMessage type representing a kafka message.
+// KafkaMessage type representing a kafka message.
 type KafkaMessage kafka.Message
 
 // eventHandler defines an eventHandler.
 type EventHandler interface {
-	HandleEvent(event event.DimensionsExtractedEvent) error
+	HandleEvent(event event.NewInstanceEvent) error
+}
+
+type KafkaConsumer interface {
+	Incoming() chan kafka.Message
+}
+
+// Completed defines an Producer for dimensions inserted events
+type CompletedProducer interface {
+	Completed(e event.InstanceCompletedEvent) error
+	Close(ctx context.Context)
 }
 
-// InsertedProducer defines an Producer for dimensions inserted events
-type InsertedProducer interface {
-	DimensionInserted(e event.DimensionsInsertedEvent) error
-	Closer() chan bool
-	Errors() chan error
+type ErrorEventHandler interface {
+	Handle(instanceID string, err error, data log.Data)
 }
 
-// Consume consume incoming kafka messages delegating to the appropriate eventHandler or handling errors.
-func Consume(dimensionExtractedConsumer KafkaMessageConsumer, insertedProducer InsertedProducer, eventHandler EventHandler, exitChannel chan error) error {
+func CloseConsumer() {
+	closer <- true
+}
+
+func Consume(in KafkaConsumer, out CompletedProducer, eventHandler EventHandler, errorEventHandler ErrorEventHandler) {
 	go func() {
-		for {
+		consuming := true
+		for consuming {
 			select {
-			case consumedMessage := <-dimensionExtractedConsumer.Incoming():
+			case consumedMessage := <-in.Incoming():
```

Comment:
Consider pulling the code from this case into a separate function, just to keep the noise out of the select statement.

Reply:
done.